File: 02.CSharpPartTwo/06_StringsAndTextProcessing/StringsAndTextProcessing/12ParseURL/ParseURL.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*Problem 12. Parse URL

Write a program that parses an URL address given in the format: [protocol]://[server]/[resource] and extracts from it the [protocol], [server] and [resource] elements.
 */

namespace _12ParseURL
{
    class ParseURL
    {
        static void Main(string[] args)
        {
            //For console input:
            //string url = Console.ReadLine();

            string url = "http://telerikacademy.com/Courses/Courses/Details/212";

            string protocolSeparator = "://";

            int indexOfProtocolSeparator = url.IndexOf(protocolSeparator);

            string protocol = url.Substring(0, indexOfProtocolSeparator);

            int indexOfServerStart = indexOfProtocolSeparator + protocolSeparator.Length;

            char serverSeparator = '/';

            int indexOfServerSeparator = url.IndexOf(serverSeparator, indexOfServerStart);

            string server = url.Substring(indexOfServerStart, indexOfServerSeparator - indexOfServerStart);

            string resource = url.Substring(indexOfServerSeparator, url.Length - indexOfServerSeparator);

            Console.WriteLine("[protocol] = {0}", protocol);
            Console.WriteLine("[server] = {0}", server);
            Console.WriteLine("[resource] = {0}", resource);
        }
    }
}

File: 06_JavaScript/09_Strings/Strings/07.ParseURL/ParseURL.js
var example = 'http://telerikacademy.com/Courses/Courses/Details/239';

function parseURL(input) {
    var i = 0,
        protocolAsString = '',
        serverAsString = '',
        resourceAsString = '',
        len = input.length;

    while (input[i].localeCompare(':')) {
        protocolAsString += input[i];
        i += 1;
    }
    i += 3;

    while (input[i].localeCompare('/')) {
        serverAsString += input[i];
        i += 1;
    }

    while (i < len) {
        resourceAsString += input[i];
        i += 1;
    }

    return {protocol: protocolAsString, server: serverAsString, resource: resourceAsString};
}

console.log(parseURL(example));